composer/utils/inference.py

```diff
@@ -0,0 +1,115 @@
+            A copy of the model is made and export related transforms work on this copy.
+        target_format (str):  format to export to. For example, torchscript, onnx, etc.
+        store_path: (str): The path string for storing the exported model.
+        store_object_store (ObjectStore, optional): If the ``path`` is in an object store
+            (i.e. AWS S3 or Google Cloud Storage), an instance of
+            :class:`~.ObjectStore` which will be used
+            to store the exported model. Otherwise, if the store_path is a local filepath, set to ``None``.
+        sample_input (Union[torch.Tensor, Sequence[torch.Tensor]], optional): Example model inputs used for tracing.
+        surgery_algs (Union[Callable, Sequence[Callable]], optional): Algorithms that should be applied to the model
+            before loading a checkpoint. Each should be callable that takes a model and returns modified model.
+        surgery_algs (Union[Callable, Sequence[Callable]], optional): transformations (usually optimizations) that should be applied to the model
+            before loading a checkpoint. Each should be a callable that takes a model and returns a modified model.
+        load_path (str): The path string to an existing checkpoint file.
+            It can be a path to a file on the local disk, a URL, or if ``load_object_store`` is set, the object name
+            for a checkpoint in a cloud bucket. For example, my_model/ep1-rank0.tar
+        load_object_store (ObjectStore, optional): If the ``path`` is in an object store
+            (i.e. AWS S3 or Google Cloud Storage), an instance of
+            :class:`~.ObjectStore` which will be used to retreive the checkpoint.
+            Otherwise, if the checkpoint is a local filepath, set to ``None``. (default: ``None``)
+
+    Returns:
+        None
+    """
+    if target_format not in ['torchscript', 'onnx']:
+        raise ValueError(
+            f'Currenlty only torchscript and onnx as target_format are supported. Received {target_format}.')
+
+    # Only rank0 exports the model
+    if dist.get_global_rank() != 0:
+        return
+
+    # make a copy of the model so that we don't modify the original model
+    model = copy.deepcopy(model)
+
+    # Apply surgery algorithms in the given order
+    for alg in ensure_tuple(surgery_algs):
+        model = alg(model)
+
+    if load_path is not None:
+        # download checkpoint and load weights only
+        tempdir_ctx = tempfile.TemporaryDirectory()
+        with tempdir_ctx as tempdir:
+            #rank_0_checkpoint_filepath = os.path.join(tempdir, 'rank0_checkpoint')
+            #get_file(destination=rank_0_checkpoint_filepath, path=load_path, object_store=load_object_store)
+            #state_dict = torch.load(composer_states_filepath, map_location='cpu')
+            #model.load_state_dict(state_dict['model'], strict=False)
+            pass
```

Comment:
Probably this is just WIP code, but gonna point out that this needs to be uncommented (or replaced)

Reply:
Yes. This was WIP. Fixed it.